### ISSUE TITLE
java.lang.NegativeArraySizeException

### ISSUE THREAD

**Issue by tinfoilhat** [tinfoilhat/tinfoil-sms#25]
``` bash
java.lang.NegativeArraySizeException
at org.spongycastle.crypto.engines.IESEngine.decryptBlock(IESEngine.java:178)
at org.spongycastle.crypto.engines.IESEngine.processBlock(IESEngine.java:310)
at com.tinfoilsms.crypto.ECEngine.processBlock(ECEngine.java:184)
at com.tinfoil.sms.crypto.Encryption.decrypt(Encryption.java:155)
at com.tinfoil.sms.utility.MessageReceiver.onReceive(MessageReceiver.java:169)
at android.app.ActivityThread.handleReceiver(ActivityThread.java:1994)
at android.app.ActivityThread.access$2400(ActivityThread.java:132)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1093)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:143)
at android.app.ActivityThread.main(ActivityThread.java:4196)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:507)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:839)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:597)
at dalvik.system.NativeStart.main(Native Method)
java.lang.NegativeArraySizeException
at org.spongycastle.crypto.engines.IESEngine.decryptBlock(IESEngine.java:178)
at org.spongycastle.crypto.engines.IESEngine.processBlock(IESEngine.java:310)
at com.tinfoilsms.crypto.ECEngine.processBlock(ECEngine.java:184)
at com.tinfoil.sms.crypto.Encryption.decrypt(Encryption.java:155)
at com.tinfoil.sms.utility.MessageReceiver.onReceive(MessageReceiver.java:169)
at android.app.ActivityThread.handleReceiver(ActivityThread.java:1994)
at android.app.ActivityThread.access$2400(ActivityThread.java:132)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1093)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:143)
at android.app.ActivityThread.main(ActivityThread.java:4196)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:507)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:839)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:597)
at dalvik.system.NativeStart.main(Native Method)

```

**Comment by gnu-user** [comment]
This issue should now be resolved with [Orwell version 1.0](https://github.com/gnu-user/orwell/tags), however I'm leaving it here until we know more about the initial cause of this error. Why are potentially null/invalid messages being processed for decryption in the first place? I've added more logging to this bit of code in the app to try and get to the bottom of it.

**Comment by gnu-user** [comment]
This issue appears to only be occurring in the event that someone untrusts a contact and then sends them a short message (less than size of MAC). The receiver would try to decrypt it, expecting it to be encrypted resulting in a NegativeArraySizeException, this is now a properly handled exception and is reported as a decryption error in the app.
